integrator.py

```diff
@@ -214,15 +220,21 @@ def train_with_context(self, context: np.ndarray, batch_size=100, lr=None, point
 
             # --------------- START TODO compute loss ---------------
             y = batch_y.to(self.device)
-            mean = torch.mean(y / absdet)
-            var = torch.var(y / absdet)
-            y = (y / mean).detach()
+            y = y + np.finfo(np.float32).eps
+
+            log_y = torch.log(y)
+            log_absdet = torch.log(absdet)
+            #mean = torch.mean(y / absdet)
+            mean = torch.mean(-log_absdet - log_y)
+            var = torch.var(y * absdet)
+            #var = torch.var(y / absdet)
+            #y = (y / mean).detach()
 
             # Backprop #
-            loss = self.loss_func(y, absdet)
+            #loss = self.loss_func(y, absdet)
+            loss = mean
```

Comment:
Do we use mean instead of loss_func?

Reply:
In this case, yes, since the loss functions use the inverse determinant. This'll need to be fixed